### PR TITLE
adds alpha support for otel

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Adds otel hookins for clues.  This adds the following:

1 - an optional initializer for users who want to run otel support.
```go
clues.Initialize(ctx)
````

2 - Span creation.
```go
ctx := clues.AddSpan(ctx, "spanName", "with_foo", "bar")
defer clues.CloseSpan(ctx)
````

3 - Automatic otel span attribute addition when Adding to clues.
4 - Automatic otel span logging when using Clog.

### CODE REVIEW

**Comment by ryanfkeepers** [comment]
this has now been through a manual test and all of the standard support has been worked out.

File: clues_test.go [hunk 1]
```diff
@@ -32,8 +32,8 @@ func mustEquals[K comparable, V any](
 ) {
 	e, g := toMSS(expect), toMSS(got)
 
-	if hasCluesTrace && len(g) > 0 {
-		if _, ok := g["clues_trace"]; !ok {
+	if len(g) > 0 {
+		if _, ok := g["clues_trace"]; !ok && hasCluesTrace {
```
Comment (ryanfkeepers) [inline clues_test.go:36]: automatic clues traces have been removed in favor of following span patterns.  So this moved to reflect that change.  Also, you'll see a lot of `true -> false` changes in the tests; those are all to match this change.  False, in this case, means "dont expect to see a clues_trace value in the map".

File: fmt.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package clues
```
Comment (ryanfkeepers) [inline fmt.go:1]: moved here from datanode.go.  The only additional change is the addition of the exported Fmt func, which is consumed by Clog.

File: datanode.go [hunk 3]
```diff
@@ -350,6 +399,75 @@ func setNodeInCtx(ctx context.Context, dn *dataNode) context.Context {
...
+	spawn.span = span
+
+	return ctx, spawn
+}
+
+// closeSpan closes the otel span and removes it span from the data node.
+// If no span is present, no ops.
+func (dn *dataNode) closeSpan(ctx context.Context) *dataNode {
+	if dn == nil || dn.span == nil {
+		return dn.spawnDescendant()
```
Comment (pandeyabs) [inline datanode.go:431]: Same concern here about returning a spawn instead of `dn`.

File: datanode.go [hunk 4]
```diff
@@ -350,6 +399,75 @@ func setNodeInCtx(ctx context.Context, dn *dataNode) context.Context {
...
+// is nil (such as if otel wasn't initialized or no span has been generated),
+// this call no-ops.
+func (dn *dataNode) addSpanAttributes(
+	values map[string]any,
+) {
+	if dn == nil || dn.span == nil {
+		return
+	}
+
+	if len(values) == 0 {
```
Comment (pandeyabs) [inline datanode.go:452]: nit: This check is redundant.

**Review by pandeyabs — APPROVED** [review]
Looks good!

File: otel.go [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+// local server instance. If it does not return an error, make sure
+// to call the client.Close() method for proper cleanup.
+// The service name is used to match traces across backends.
+func newOTELClient(
+	ctx context.Context,
+	serviceName string,
+	config OTELConfig,
+) (*otelClient, error) {
+	// -- Resource
+	resource, err := resource.New(ctx, resource.WithAttributes(
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply otel.go:78]: *gasp* where are my linters?

File: otel.go [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+	if err != nil {
+		return nil, WrapWC(ctx, err, "creating otel resource")
+	}
+
+	// -- Exporter
+
+	conn, err := grpc.NewClient(
+		config.GRPCEndpoint,
+		// Note the use of insecure transport here. TLS is recommended in production.
+		grpc.WithTransportCredentials(insecure.NewCredentials()))
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply otel.go:89]: Yep, that will be necessary.

File: otel.go [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+	// -- TracerProvider
+
+	// Register the trace exporter with a TracerProvider, using a batch
+	// span processor to aggregate spans before export.
+	batchSpanProcessor := sdkTrace.NewBatchSpanProcessor(exporter)
+
+	tracerProvider := sdkTrace.NewTracerProvider(
+		sdkTrace.WithResource(resource),
+		sdkTrace.WithSampler(sdkTrace.AlwaysSample()),
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply otel.go:107]: Sampling rates, absolutely.  Other configurations can creep into exported configuration as needed.

File: datanode.go [hunk 8]
```diff
@@ -350,6 +399,75 @@ func setNodeInCtx(ctx context.Context, dn *dataNode) context.Context {
...
+// Attrs can be added to the span with addSpanAttrs.  This span will
+// continue to be used for that purpose until replaced with another
+// span, which will appear in a separate context (and thus a separate,
+// dataNode).
+func (dn *dataNode) addSpan(
+	ctx context.Context,
+	name string,
+) (context.Context, *dataNode) {
+	if dn == nil || dn.otel == nil {
+		return ctx, dn.spawnDescendant()
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply datanode.go:416]: ah, that's just a bug.  Good catch.

**Pull request merged**